executor/server.go

```diff
@@ -50,6 +62,7 @@ func NewServer(cfg *Config, ctx *test.Context) *Server {
 		testCtx:     ctx,
 		cliUpdateCh: make(chan []string),
 	}
+	s.discoveryConnector = s.connectToEtcdDiscovery
```

Comment:
Why delegate this function instead of using it directly?

Reply:
Use a function variable here to make it easy for unit test.